src/selectors/profile.js

```diff
@@ -347,6 +347,151 @@ function _createCounterSelectors(counterIndex: CounterIndex) {
 export const getIPCMarkerCorrelations: Selector<IPCMarkerCorrelations> =
   createSelector(getThreads, correlateIPCMarkers);
 
+/**
+ * Returns an InnerWindowID -> Page map, so we can look up the page from inner
+ * window id quickly. Returns null if there are no pages in the profile.
+ */
+export const getInnerWindowIDToPageMap: Selector<Map<
+  InnerWindowID,
+  Page,
+> | null> = createSelector(getPageList, (pages) => {
+  if (!pages) {
+    // Return null if there are no pages.
+    return null;
+  }
+
+  const innerWindowIDToPageMap: Map<InnerWindowID, Page> = new Map();
+  for (const page of pages) {
+    innerWindowIDToPageMap.set(page.innerWindowID, page);
+  }
+
+  return innerWindowIDToPageMap;
+});
```

Comment:
~~The pages list is usually very short, therefore I wonder if a simple loop over all pages would be good enough instead of using a map and memoizing it (which both have costs too) and introduce some complexity. What do you think?~~

__Update:__ oh I see you mostly moved that and that it's used in a lot of places, I guess this ship has sailed now :-)

Reply:
Yeah, we are also using it in various places so it made sense at the time. I think it's fine to keep it still.